### PR TITLE
[SPARK-7808] [ML] add package doc for ml.feature

### PR DESCRIPTION
**Pull request by mengxr**
This PR adds a short description of `ml.feature` package with code example. The Java package doc will come in a separate PR. @jkbradley

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/feature/package.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ *   val model = pipeline.fit(df)
+ *
+ *   // save transformed features with raw data
+ *   model.transform(df)
+ *     .select("id", "text", "rating", "features")
+ *     .write.format("parquet").save("/output/path")
+ * }}}
+ *
+ * Some feature transformers implemented in MLlib are inspired by those implemented in scikit-learn.
+ * The major difference is that most scikit-learn's feature transformers operate eagerly on the
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/ml/feature/package.scala]: "most scikit-learn's" --> "most scikit-learn"

**Comment by jkbradley** [comment]
Include link to online API doc?

**Comment by jkbradley** [comment]
Other than that, LGTM
(I generated it, and ran the example)

**Comment by mengxr** [comment]
We don't have a dedicated user guide section for `ml.feature` yet. So I didn't provide a link.

**Comment by mengxr** [comment]
Merged into master and branch-1.5.

**Comment by jkbradley** [comment]
I was suggesting the API guide, not the user guide (but we actually do have a user guide section as well: ml-feature.html)

**Pull request closed without merging**